standard/statements.md

```diff
@@ -366,7 +366,7 @@ implicitly_typed_local_variable_declarator
     ;
 ```
 
-An *implicity_typed_local_variable_declaration* introduces a single local variable, *identifier*. The *expression* or *variable_reference* shall have a compile-time type, `T`. The first alternative declares a variable with type `T` and an initial value of *expression*. The second alternative declares a ref variable with type `ref T` and an initial value of `ref` *variable_reference*.
+An *implicity_typed_local_variable_declaration* introduces a single local variable, *identifier*. The *expression* or *variable_reference* shall have a compile-time type, `T`. The first alternative declares a variable with an initial value of *expression*; its type `T?` when `T` is a reference type and `T` when `T` is a value type. The second alternative declares a ref variable with an initial value of `ref` *variable_reference*; its type is `ref T?` when `T` is a reference type and `ref T` when `T` is a value type.
```

Comment:
What about when the expression already has a type which is a nullable reference type? For example:

```csharp
string? x = null;
var y = x;
```

Doesn't this paragraph suggest that `y` has a type of `string??` ?

The `ref` version of this feels dangerous to me - but seems to be "correct". This code compiles with no warnings, and throws at runtime:

```csharpstring x = "";
ref var y = ref x;
y = null;

Console.WriteLine(x.Length);
```